### ISSUE TITLE
clicking on inline links reloads the page

### ISSUE THREAD

**Issue by johanneswilm** [WordPress/gutenberg#10117]
**Describe the bug**
Internal links should just make the selection jump to the link's target. Instead they make the page reload.

**To Reproduce**
1. Create headline with anchor value "myanchor"
2. Create a link with target "#myanchor"
3. Click on link within the editor
4. The page reloads

**Expected behavior**
the caret should jump to the headline with anchor value "myanchor" and the page should not reload.

**Comment by designsimply** [comment]
Tested and confirmed that if I create a link to `#myanchor` and click the link in the floating link toolbar, it will open the editor in a new tab or window.

![screen shot 2018-09-24 at 2 26 12 pm](https://user-images.githubusercontent.com/1119271/45977141-ec804980-c005-11e8-9933-e429a814c96f.png)

**Comment by talldan** [comment]
Currently still an issue.

**Comment by gwwar** [comment]
Verified that this is still an issue 👍 

https://user-images.githubusercontent.com/1270189/112531699-1d6f9f80-8d65-11eb-8672-059f0136d22e.mp4

**Comment by gwwar** [comment]
While implementing this, we may also want to keep in mind allowing browser link copy -> pasting a link on text, retaining the right link

https://user-images.githubusercontent.com/1270189/112535428-86591680-8d69-11eb-8be1-55395c83bd0b.mp4

See https://github.com/WordPress/gutenberg/issues/28115#issuecomment-807366947

**Comment by arthur791004** [comment]
Hi all, I think the current expected behavior described above is great but I have another thoughts. There is an external link icon which indicates it would open the new tab after clicking. Therefore, as a user, I might expect that when I clicked it, it would open the new tab and scroll to anchor. However, **the page should not be edit mode** because I want to test the behavior in other people's point rather than mine. 

In short, the relative anchor link should have the same behavior with absolute link in this demo video https://github.com/WordPress/gutenberg/issues/10117#issuecomment-807319139 

Does it make sense?

**Comment by skorasaurus** [comment]
still an issue on gutenberg 13.0.0.

**Comment by carolinan** [comment]
Yes, still an issue. 
Mac OS, Chrome, WordPress 6.0 and Gutenberg trunk (13.6.x).

https://user-images.githubusercontent.com/7422055/177924891-a14c2f33-81d0-42e2-9a64-4c39eaa555b6.mov

I wonder if it would be better (and certainly easier?) to prevent actions when clicking on a link that starts with `#`, rather than try to move focus to the link target in the editor.

**Comment by carolinan** [comment]
I wonder why it is labeling local links -non anchor links to existing content on the same install -as external 🤔

**Comment by gwwar** [comment]
I took a quick look at this and I wasn't sure what the ideal behavior is.

* If we'd like to open the url to a preview page with the anchor selected, one problem to work through is that the link-preview component exists in the block-editor package, and should not have any awareness of WordPress details (such as where a published post will be) . See the [Modularity page](https://developer.wordpress.org/block-editor/explanations/architecture/modularity/) for more details.
* We could make the behavior jump to the linked anchor block in the editor, but it feels a bit expensive to do this. Primarily we'd need to search over all blocks to see if we save an anchor attribute (or alternatively parse content to see if we have a matching id). 

Eg very roughly something like the following 
```
// likely too expensive to be worth it.
const blocks = wp.data.select('core/block-editor').getBlocks()
// check each block to see if it has an anchor attribute, or has a matching id in content
wp.data.dispatch( 'core/block-editor' ).selectBlock(clientId)
```

I wonder if it makes the most sense to simply do nothing when we click on this type of anchor link.

Let me know if folks think of other options or ideas.

**Comment by carolinan** [comment]
Yes "do nothing" was implemented in https://github.com/WordPress/gutenberg/pull/42259.

**Comment by youknowriad** [comment]
Going to close this as the "do nothing" option above seems the most sensible and has been implemented.
